tests/multi_objective_tests/samplers_tests/test_motpe.py

```diff
@@ -63,545 +75,7 @@ def test_sample_independent_seed_fix() -> None:
-
-    assert len(set(suggestions)) == 1
-
-
-def test_get_observation_pairs() -> None:
-    def objective(trial: optuna.multi_objective.trial.Trial) -> Tuple[float, float]:
-        trial.suggest_int("x", 5, 5)
-        return 5.0, 5.0
-
-    sampler = MOTPEMultiObjectiveSampler(seed=0)
-    study = optuna.multi_objective.create_study(
-        directions=["minimize", "maximize"], sampler=sampler
-    )
-    study.optimize(objective, n_trials=5)
-
-    assert _motpe._get_observation_pairs(study, "x") == (
-        [5.0, 5.0, 5.0, 5.0, 5.0],
-        [[5.0, -5.0], [5.0, -5.0], [5.0, -5.0], [5.0, -5.0], [5.0, -5.0]],
-    )
-    assert _motpe._get_observation_pairs(study, "y") == (
-        [None, None, None, None, None],
-        [[5.0, -5.0], [5.0, -5.0], [5.0, -5.0], [5.0, -5.0], [5.0, -5.0]],
-    )
-
-
-def test_calculate_nondomination_rank() -> None:
-    # Single objective
-    test_case = np.asarray([[10], [20], [20], [30]])
-    ranks = list(_motpe._calculate_nondomination_rank(test_case))
-    assert ranks == [0, 1, 1, 2]
-
-    # Two objectives
-    test_case = np.asarray([[10, 30], [10, 10], [20, 20], [30, 10], [15, 15]])
-    ranks = list(_motpe._calculate_nondomination_rank(test_case))
-    assert ranks == [1, 0, 2, 1, 1]
-
-    # Three objectives
-    test_case = np.asarray([[5, 5, 4], [5, 5, 5], [9, 9, 0], [5, 7, 5], [0, 0, 9], [0, 9, 9]])
-    ranks = list(_motpe._calculate_nondomination_rank(test_case))
-    assert ranks == [0, 1, 0, 2, 0, 1]
-
-
-def test_solve_hssp() -> None:
-    sampler = MOTPEMultiObjectiveSampler(seed=0)
-
-    random.seed(128)
-
-    # Two dimensions
-    for i in range(8):
-        subset_size = int(random.random() * i) + 1
-        test_case = np.asarray([[random.random(), random.random()] for _ in range(8)])
-        r = 1.1 * np.max(test_case, axis=0)
-        truth = 0.0
-        for subset in itertools.permutations(test_case, subset_size):
-            truth = max(truth, sampler._compute_hypervolume(np.asarray(subset), r))
-        indices = sampler._solve_hssp(test_case, np.arange(len(test_case)), subset_size, r)
-        approx = sampler._compute_hypervolume(test_case[indices], r)
-        assert approx / truth > 0.6321  # 1 - 1/e
-
-    # Three dimensions
-    for i in range(8):
-        subset_size = int(random.random() * i) + 1
-        test_case = np.asarray(
-            [[random.random(), random.random(), random.random()] for _ in range(8)]
-        )
-        r = 1.1 * np.max(test_case, axis=0)
-        truth = 0
-        for subset in itertools.permutations(test_case, subset_size):
-            truth = max(truth, sampler._compute_hypervolume(np.asarray(subset), r))
-        indices = sampler._solve_hssp(test_case, np.arange(len(test_case)), subset_size, r)
-        approx = sampler._compute_hypervolume(test_case[indices], r)
-        assert approx / truth > 0.6321  # 1 - 1/e
+        _ = sampler.sample_independent(study, trial, "param-a", dist)
```

Comment:
I'd like to know the reason this test is not removed and only invoke `sample_independent`.

Reply:
Looks like it's to test the _create_study which he had to create for this wrapper.

Reply:
Thanks for the question. I agree with @Crissman. I confirmed the behavior of the added `_create_study` and `_create_trial` by making sure that the public method, `sample_independent`, worked.

Another way is to test `_create_study` and `_create_trial` directly, but I prefer to test the public method here (not a strong opinion). What do you think? @himkt

Reply:
Thank you for clarifying. It makes sense. And now tests is improved (in 80a724d).

[nits]
Is it good to check `sampler._motpe_sampler.sample_independent` is called instead of `sampler.sample_independent`?
It feels a little natural for me to use `mock.call_count` with a wrapped object if it tests whether a wrapper works.

```diff
diff --git a/tests/multi_objective_tests/samplers_tests/test_motpe.py b/tests/multi_objective_tests/samplers_tests/test_motpe.py
index 4426b10a..1c925c1b 100644
--- a/tests/multi_objective_tests/samplers_tests/test_motpe.py
+++ b/tests/multi_objective_tests/samplers_tests/test_motpe.py
@@ -8,6 +8,7 @@ import pytest
 import optuna
 from optuna import multi_objective
 from optuna.multi_objective.samplers import MOTPEMultiObjectiveSampler
+from optuna.samplers import MOTPESampler


 class MockSystemAttr:
@@ -61,6 +62,10 @@ def test_sample_independent() -> None:
         y = trial.suggest_float("y", 0, 1)
         return x, y

-    with patch.object(sampler, "sample_independent", wraps=sampler.sample_independent) as mock:
+    with patch.object(
+        MOTPESampler,
+        "sample_independent",
+        wraps=sampler._motpe_sampler.sample_independent,
+    ) as mock:
         study.optimize(_objective, n_trials=10)
         assert mock.call_count == 20
```